account_invoice_origin_link/views/account_move_views.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2022 ForgeFlow S.L.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
+<odoo>
+    <record id="view_move_form" model="ir.ui.view">
+        <field name="name">account.move.form (account_move_reference_link)</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <field name="invoice_origin" position="replace">
+                <field
+                    name="invoice_origin"
+                    string="Source Document"
+                    attrs="{'invisible': ['|',('invoice_origin', '=', False),('origin_reference', '!=', False)]}"
+                />
+                <field
+                    name="origin_reference"
+                    attrs="{'invisible': [('origin_reference', '=', False)]}"
```

Comment:
Just a question: Suppose `invoice_origin` changes with a valid origin. Then by the _compute_source_doc_ref, it sets `origin_reference` to False. Then how do you manually set the corresponding new origin_reference or is it left to False forever?

Reply:
Origin reference is meant to be filled based on the invoice_origin. In the case of the module we are adding, it will be based on the related sales order, but you could add new modules to make it rely on other related documents